tchap_username_email/__init__.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Optional
+
+import attr
+from synapse.module_api import ModuleApi
+
+AUTH_TYPE_EMAIL = "m.login.email.identity"
+
+
+@attr.s(auto_attribs=True, frozen=True)
+class TchapUsernameEmailConfig:
+    extract_from_email: bool
+
+
+class TchapUsernameEmail:
+    def __init__(self, config: TchapUsernameEmailConfig, api: ModuleApi):
+        # Keep a reference to the config and Module API
+        self._api = api
```

Comment:
Why do we keep a reference to the API? We seem to only use it here.

Reply:
Good point, that's mostly a remain of the cookiecutter template. Will remove.